connectors/sqs/src/test/java/io/camunda/connector/inbound/SqsQueueConsumerTest.java

```diff
@@ -0,0 +1,142 @@
+    queue.setPollingWaitTime("1");
+
+    properties.setQueue(queue);
+
+    isQueueConsumerActive = new AtomicBoolean(true);
+
+    consumer = new SqsQueueConsumer(sqsClient, properties, context, isQueueConsumerActive);
+  }
+
+  @Test
+  void run_shouldActivate() throws InterruptedException {
+    // given
+    when(sqsClient.receiveMessage(any(ReceiveMessageRequest.class)))
+        .thenReturn(receiveMessageResult);
+    when(receiveMessageResult.getMessages()).thenReturn(messages);
+    when(messages.iterator()).thenReturn(Collections.singletonList(message).iterator());
+    when(context.correlate(message)).thenReturn(result);
+    when(result.isActivated()).thenReturn(true);
+    // when
+    Thread thread =
+        new Thread(
+            () -> {
+              consumer.run();
+            });
+    thread.start();
+    isQueueConsumerActive.set(false);
+    thread.join();
+    // then
+    verify(sqsClient).receiveMessage(any(ReceiveMessageRequest.class));
+    verify(context).correlate(message);
+  }
+
+  @Test
+  void run_shouldActivateWithAttributes() throws InterruptedException {
+    // given
+    List<String> attributeNames = Collections.singletonList("attribute");
+    List<String> messageAttributeNames = Collections.singletonList("attribute");
+    queue.setAttributeNames(attributeNames);
+    queue.setMessageAttributeNames(messageAttributeNames);
+    when(sqsClient.receiveMessage(requestArgumentCaptor.capture()))
+        .thenReturn(receiveMessageResult);
+    when(receiveMessageResult.getMessages()).thenReturn(messages);
+    when(messages.iterator()).thenReturn(Collections.singletonList(message).iterator());
+    when(context.correlate(message)).thenReturn(result);
+    when(result.isActivated()).thenReturn(true);
+    // when
+    Thread thread =
+        new Thread(
+            () -> {
+              consumer.run();
+            });
+    thread.start();
+    isQueueConsumerActive.set(false);
+    thread.join();
+    // then
+    verify(sqsClient).receiveMessage(any(ReceiveMessageRequest.class));
+    verify(context).correlate(message);
+    ReceiveMessageRequest receiveMessageRequest = requestArgumentCaptor.getValue();
+    assertThat(receiveMessageRequest.getAttributeNames()).isEqualTo(attributeNames);
+    assertThat(receiveMessageRequest.getMessageAttributeNames()).isEqualTo(messageAttributeNames);
+  }
+
+  @Test
+  void consumeRun_withNoResults() throws InterruptedException {
+    // given
+    when(sqsClient.receiveMessage(any(ReceiveMessageRequest.class)))
+        .thenReturn(receiveMessageResult);
+    when(receiveMessageResult.getMessages()).thenReturn(Collections.emptyList());
+    // when
+    Thread thread =
+        new Thread(
+            () -> {
+              consumer.run();
+              verify(sqsClient).receiveMessage(any(ReceiveMessageRequest.class));
+            });
+    thread.start();
+    // then
+    CountDownLatch waiter = new CountDownLatch(1);
+    waiter.await(1, TimeUnit.SECONDS);
```

Comment:
We can eliminate the waiting time entirely. Since you use a do-while loop, that will execute at least once regardless of the value of isQueueConsumerActive, you can set that to false at the beginning to execute your do-while loop only once. Now you only have to wait for the thread to finish. Luckily we have [Thread.join()](https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#join--) method.
So you can write this:
```
    thread.start();
    isQueueConsumerActive.set(false);

    // then
    thread.join();
    verifyNoInteractions(context);
```
This way we do not block the execution of the unit tests. 
Related: I am still not convinced using a global boolean variable is the best approach to stop the loop. see: https://github.com/camunda/connectors-bundle/pull/533#discussion_r1196483966